node/batcher/batcher_role.go

```diff
@@ -216,6 +217,13 @@ func (b *BatcherRole) resubmitPendingBAFs(state *state.State, prevPrimary types.
 		if baf.Shard() == b.Shard && baf.Signer() == b.ID && baf.Primary() == prevPrimary {
 			b.Logger.Debugf("found pending BAF signed by me (id: %d) from prev primary: %d ; %s", b.ID, prevPrimary, baf.String())
 			batch := b.Ledger.RetrieveBatchByNumber(baf.Primary(), uint64(baf.Seq()))
+			if batch == nil {
+				b.Logger.Infof("My ledger hight is %d", b.Ledger.Height(b.ID))
```

Comment:
Format these diagnostics such that first you announce "Error: <what>", then emit extra info, then panic.